### ISSUE TITLE
Instances for `indexed-traversable`

### ISSUE THREAD

**Issue by konsumlamm** [lspitzner/pqueue#80]
[`indexed-traversable`](https://hackage.haskell.org/package/indexed-traversable) provides the `FunctorWithIndex`, `FoldableWithIndex` and `TravresableWithIndexed` classes, all of which could be easily implemented by `MinPQueue` and `MaxPQueue` (using existing functions). The extra dependencies aren't too bad imo (pretty much only boot libraries afaict).

**Comment by treeowl** [comment]
Yes, with the caveat that the folding and traversing are unordered.

**Comment by konsumlamm** [comment]
> Yes, with the caveat that the folding and traversing are unordered.

No, I'm talking about `MinPQueue` (from `Data.PQueue.Prio.Min`) and `MaxPQueue` (from `Data.PQueue.Prio.Max`), which both have `mapWithKey`, `foldrWithKey`, `foldlWithKey` and `traverseWithKey` (which are all ordered).

> How bad are the transitive dependencies?

For GHC >= 8.0 it's just `array`, `base`, `containers` and `transformers` (which have no relevant further dependencies). For GHC 7.10, it's quite a few more dependencies, so perhaps we should drop 7.10 support when adding a dependency on `indexed-traversable`.

**Comment by treeowl** [comment]
Extra compat deps are fine.
